File: python/pe5.py
def gcd(a, b):
    if a < b: return gcd(b, a)
    if b == 0: return a
    return gcd(b, a%b)

def lcm(a, b):
    return a*b/gcd(a,b)

print reduce(lcm, range(1, 20))

File: scala/pe5.scala
object ProjectEuler5 {
  def gcd(x: Int, y: Int): Int = if (x == 0) y else gcd(y % x, x)

  def lcm(x: Int, y: Int): Int = x * y / gcd(x, y)

  def main(args: Array[String]) = {
    println( (1 to 20).reduceLeft(lcm) )
  }
}
